### PR TITLE
New react redux

### PR DESCRIPTION
**Pull request by max-degterev**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,40 +23,40 @@
...
-    "@babel/core": "^7.4.3",
+    "@babel/cli": "^7.4.4",
+    "@babel/core": "^7.4.5",
     "@babel/plugin-proposal-json-strings": "^7.0.0",
-    "@babel/plugin-transform-runtime": "^7.4.3",
+    "@babel/plugin-transform-runtime": "^7.4.4",
     "@babel/plugin-transform-strict-mode": "^7.0.0",
-    "@babel/preset-env": "^7.4.3",
+    "@babel/preset-env": "^7.4.5",
     "@babel/preset-react": "^7.0.0",
```
Comment (atomiomi) [inline package.json:32]: do we need react transform here?  There is only one component render in each file that can be replaced with `React.createElement`

File: src/utils/index.es [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React, { forwardRef } from 'react';
+
+export const getDisplayName = (wrapper, Component) => {
+  const name = Component.displayName || Component.name || 'Component';
+  return `${wrapper}(${name})`;
+};
+
+export const getForwardedComponent = (displayName, Component) => {
+  const WrappedComponent = (props, ref) => <Component {...props} forwardedRef={ref} />;
+  WrappedComponent.displayName = displayName;
```
Comment (atomiomi) [inline src/utils/index.es:10]: get `displayName` from `Component`? it's different only in one case. you can reassign it there

File: src/adaptive/index.es [hunk 3]
```diff
@@ -26,9 +27,16 @@ const makeAdaptive = (Component) => (
 
     render() {
       const { isMobile } = this.state;
-      return <Component {...this.props} mobile={isMobile} />;
+      const { forwardedRef, ...props } = this.props;
```
Comment (atomiomi) [inline src/adaptive/index.es:30]: add `forwardRef` to `propTypes`?

File: src/merge_methods/index.es [hunk 4]
```diff
@@ -1,13 +1,14 @@
 import React, { PureComponent } from 'react';
+import { getDisplayName } from '../utils';
 
 
-export default (methods, WrappedComponent) => {
-  const displayName = WrappedComponent.displayName || WrappedComponent.name || 'Component';
+const mergeMethods = (methods, Component) => {
```
Comment (atomiomi) [inline src/merge_methods/index.es]: do we need this with forwardRef?

File: src/utils/index.es [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React, { forwardRef } from 'react';
+
+export const getDisplayName = (wrapper, Component) => {
```
Comment (atomiomi) [inline src/utils/index.es:3]: `getProxyDisplayName | getWrapperDisplayName`?

File: src/merge_methods/index.es (hunk 4 above)
Comment (atomiomi) [inline src/merge_methods/index.es:5]: do we still need this with forwarfRef?

File: src/adaptive/index.es [hunk 6]
```diff
@@ -1,9 +1,10 @@
 import React, { PureComponent } from 'react';
 import eventproxy from 'nebenan-helpers/lib/eventproxy';
 import { getMedia, media } from 'nebenan-helpers/lib/dom';
+import { getDisplayName, getForwardedComponent } from '../utils';
 
 
-const makeAdaptive = (Component) => (
+const makeAdaptive = (Component, options = {}) => {
```
Comment (atomiomi) [inline src/adaptive/index.es:7]: probably we can replace this with hook

File: package.json (hunk 1 above)
Comment (max-degterev) [reply package.json:32]: Build time dependency - doesnt matter

File: src/adaptive/index.es (hunk 6 above)
Comment (max-degterev) [reply src/adaptive/index.es:7]: Next release tho

File: src/merge_methods/index.es (hunk 4 above)
Comment (max-degterev) [reply src/merge_methods/index.es:5]: Needed for merging methods from multiple components into 1. I reduced use cases to just 2 components, but no fix possible without duplicating all of the methods of wrapped ref

File: src/utils/index.es (hunk 5 above)
Comment (max-degterev) [reply src/utils/index.es:3]: Same thing tho isnt it?

File: src/utils/index.es (hunk 2 above)
Comment (max-degterev) [reply src/utils/index.es:10]: That's the wrapper component, not the original. displayName is empty

File: src/adaptive/index.es (hunk 3 above)
Comment (max-degterev) [reply src/adaptive/index.es:30]: actually there's no propTypes validation, build script strips that out anyway.

File: src/utils/index.es (hunk 5 above)
Comment (atomiomi) [reply src/utils/index.es:3]: maybe getPrefixedDisplayName?

**Pull request merged**